pkg/solana/chainreader/event_read_binding.go

```diff
@@ -0,0 +1,245 @@
+package chainreader
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/gagliardetto/solana-go"
+
+	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logpoller"
+)
+
+type eventReadBinding struct {
+	namespace, genericName string
+	codec                  types.RemoteCodec
+	modifier               commoncodec.Modifier
+	key                    solana.PublicKey
+	remapper               remapHelper
+	indexedSubkeys         map[string]string
```

Comment:
```suggestion
	indexedSubkeys         map[string]int
```
(could be int64 or uint64, etc. doesn't matter much I think? This is just going to be a number between 0 and NumDBIndexes - 1 (where NumDBIndexes is how many CREATE INDEX statements we have in the sql migration script, which determines the maximum # of subkey fields you can index.)

Currently we have NumDBIndexes = 4, but we can increase that if there's a product that needs more than 4 different fields of the same event indexed:
```
core/store/migrate/migrations/0262_add_solana_schema.sql:CREATE INDEX IF NOT EXISTS solana_logs_idx_subkey_one ON solana.logs ((subkey_values[1]));
core/store/migrate/migrations/0262_add_solana_schema.sql:CREATE INDEX IF NOT EXISTS solana_logs_idx_subkey_two ON solana.logs ((subkey_values[2]));
core/store/migrate/migrations/0262_add_solana_schema.sql:CREATE INDEX IF NOT EXISTS solana_logs_idx_subkey_three ON solana.logs ((subkey_values[3]));
core/store/migrate/migrations/0262_add_solana_schema.sql:CREATE INDEX IF NOT EXISTS solana_logs_idx_subkey_four ON solana.logs ((subkey_values[4]));
```